From 804b064e3940282a61017baa1e3ded22d829a417 Mon Sep 17 00:00:00 2001
From: muralikrishnabilla <murali.krishna.billa@intel.com>
Date: Mon, 20 Mar 2023 20:33:01 +0530
Subject: [PATCH 41/42] ui/gtk: reshow VC if it's main VC

Primary VC's window is NULL so it can't pass the null check in window
state event call back. We need to add a condition to get it
in shown state again.

Signed-off-by: muralikrishnabilla <murali.krishna.billa@intel.com>
---
 ui/gtk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ui/gtk.c b/ui/gtk.c
index 64766cce70..0f1340bb7f 100644
--- a/ui/gtk.c
+++ b/ui/gtk.c
@@ -1560,7 +1560,7 @@ static gboolean gd_window_state_event(GtkWidget *widget, GdkEvent *event,
     if (event->window_state.new_window_state & GDK_WINDOW_STATE_ICONIFIED) {
         gd_ui_hide(vc);
     } else {
-        if (vc->window) {
+        if (vc->window || vc == gd_vc_find_current(vc->s)) {
             gd_ui_show(vc);
         }
     }
-- 
2.34.1

